### PR TITLE
fix(crank): error out on timeout installing package

### LINKED ISSUE #5149: `--wait` flag on package install doesn't do anything when the wait timeout expires before the package is healthy
### What happened?
https://docs.crossplane.io/v1.14/cli/command-reference/#wait-for-package-install says that `--wait` will error out if the timeout is hit before the package becomes healthy. This is not true. It just stops waiting and reports success, whether the package is healthy or not.

### How can we reproduce it?
Install a configuration package with a large number of dependencies. The package won't report healthy until all of the dependencies are installed, but the install command will exit anyway.
```bash
crossplane xpkg install configuration ghcr.io/back-stack/showcase-configuration:latest --wait=30s && kubectl get configuration

NAME                                INSTALLED   HEALTHY   PACKAGE                                            AGE
back-stack-showcase-configuration   True        Unknown   ghcr.io/back-stack/showcase-configuration:latest   30s
```

I would expect the command to error out, as described in the docs.

### What environment did it happen in?
Crossplane version: 
1.14.4-up.1

kind v0.20.0 go1.20.5 darwin/arm64 
Client Version: v1.28.4
Server Version: v1.27.3

### PR DESCRIPTION
**Pull request by phisco**
### Description of your changes

Fixes https://github.com/crossplane/crossplane/issues/5149.

Built locally with `make build`, then:
```shell
# not existing provider
$ _output/bin/darwin_arm64/crank xpkg install provider crossplanecontrib/provider-nop:latest test --wait=1s && kubectl get providers

crossplane: error: Package did not become ready: context deadline exceeded
```
with exit code 1.

I have: 

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] ~Added or updated unit tests.~
- [ ] ~Added or updated e2e tests.~
- [ ] ~Linked a PR or a [docs tracking issue] to [document this change].~
- [x] Added `backport release-x.y` labels to auto-backport this PR.

Need help with this checklist? See the [cheat sheet].

[contribution process]: https://github.com/crossplane/crossplane/tree/master/contributing
[docs tracking issue]: https://github.com/crossplane/docs/issues/new
[document this change]: https://docs.crossplane.io/contribute/contribute
[cheat sheet]: https://github.com/crossplane/crossplane/tree/master/contributing#checklist-cheat-sheet

### CODE REVIEW

File: cmd/crank/xpkg/install.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ type installCmd struct {
 	ManualActivation     bool          `short:"m" help:"Require the new package's first revision to be manually activated."`
 	PackagePullSecrets   []string      `placeholder:"NAME" help:"A comma-separated list of secrets the package manager should use to pull the package from the registry."`
 	RevisionHistoryLimit int64         `short:"r" placeholder:"LIMIT" help:"How many package revisions may exist before the oldest revisions are deleted."`
-	Wait                 time.Duration `short:"w" default:"0s" help:"How long to wait for the package to install before returning. The command does not wait by default."`
+	Wait                 time.Duration `short:"w" default:"0s" help:"How long to wait for the package to install before returning. The command does not wait by default. Errors out if the timeout is exceeded."`
```
Comment (jbw976) [inline cmd/crank/xpkg/install.go:65]: nit wording suggestion: 
```suggestion
	Wait                 time.Duration `short:"w" default:"0s" help:"How long to wait for the package to install before returning. The command does not wait by default. Returns an error if the timeout is exceeded."`
```

**Review by jbw976 — APPROVED** [review]
LGTM @phisco! anything you can add to the PR about how you tested it and what the output looks like when this condition is hit? 🤔

**Comment by phisco** [comment]
updated the description @jbw976. The Output is not the best, but we should address it more organically in my opinion.

**Comment by jbw976** [comment]
cool! looks reasonable enough for me now 👍

**Review by negz — APPROVED** [review]

**Comment by negz** [comment]
@phisco Do you want to back port this? Just asking because you left the relevant box unticked.

**Comment by phisco** [comment]
@negz I was thinking about that, yes. I'm not sure it's actually worth it with a release due soon, I could be easily convinced it's not, WDYT?

**Pull request merged**